sycl/unittests/program/KernelRelease.cpp

```diff
@@ -0,0 +1,97 @@
+//==----------- KernelRelease.cpp --- kernel release unit test -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <detail/context_impl.hpp>
+#include <gtest/gtest.h>
+#include <helpers/PiMock.hpp>
+
+#include <iostream>
+#include <memory>
+
+using namespace cl::sycl;
+
+struct TestCtx {
+  TestCtx(context &Ctx) : Ctx{Ctx} {};
+
+  context &Ctx;
+  int KernelReferenceCount = 0;
+};
+
+std::unique_ptr<TestCtx> TestContext;
+
+pi_result redefinedProgramCreateWithSource(pi_context context, pi_uint32 count,
+                                           const char **strings,
+                                           const size_t *lengths,
+                                           pi_program *ret_program) {
+  return PI_SUCCESS;
+}
+
+pi_result
+redefinedProgramBuild(pi_program program, pi_uint32 num_devices,
+                      const pi_device *device_list, const char *options,
+                      void (*pfn_notify)(pi_program program, void *user_data),
+                      void *user_data) {
+  return PI_SUCCESS;
+}
+
+pi_result redefinedKernelCreate(pi_program program, const char *kernel_name,
+                                pi_kernel *ret_kernel) {
+  TestContext->KernelReferenceCount = 1;
+  return PI_SUCCESS;
+}
+
+pi_result redefinedKernelRetain(pi_kernel kernel) {
+  ++TestContext->KernelReferenceCount;
+  return PI_SUCCESS;
+}
+
+pi_result redefinedKernelRelease(pi_kernel kernel) {
+  --TestContext->KernelReferenceCount;
+  return PI_SUCCESS;
+}
+
+pi_result redefinedKernelGetInfo(pi_kernel kernel, pi_kernel_info param_name,
+                                 size_t param_value_size, void *param_value,
+                                 size_t *param_value_size_ret) {
+  EXPECT_EQ(param_name, PI_KERNEL_INFO_CONTEXT)
+      << "Unexpected kernel info requested";
+  auto *Result = reinterpret_cast<RT::PiContext *>(param_value);
+  RT::PiContext PiCtx =
+      detail::getSyclObjImpl(TestContext->Ctx)->getHandleRef();
+  *Result = PiCtx;
+  return PI_SUCCESS;
+}
+
+TEST(KernelReleaseTest, GetKernelRelease) {
+  unittest::PiMock Mock;
+  platform Plt = Mock.getPlatform();
+  if (Plt.is_host()) {
+    std::cerr << "Not run due to host-only environment\n";
```

Comment:
Minor. Could you explicitly search for non host platform + improve the message or add a comment why the test is skipped on the "host" environment. + please, add std::endl;

Reply:
Applied partially, improved the message and added std::endl, but as for the explicit search for a non-host platform, I think we can safely rely on the behaviour of default_selector here.

Reply:
Ok.